components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -256,12 +256,8 @@ func Run(options ...RunOption) {
 	var metricsReporter *metrics.GrpcMetricsReporter
 	if opts.RunGP {
 		cstate.MarkContentReady(csapi.WorkspaceInitFromOther)
-	} else {
-		analytics := analytics.NewFromEnvironment()
-		defer analytics.Close()
-		go analyseConfigChanges(ctx, cfg, analytics, gitpodConfigService, gitpodService)
-		go analysePerfChanges(ctx, cfg, analytics, topService, gitpodService)
-
+	} else if !cfg.isHeadless() {
+		go startAnalyze(ctx, cfg, gitpodConfigService, topService, gitpodService)
 		_, gitpodHost, err := cfg.GitpodAPIEndpoint()
```

Comment:
hm, analytics makes sense, but not sure about observability, we still want to know if we cannot stream tasks for prebuilds from the server

Reply:
I've updated it to address the comment above, could you review it again? @akosyakov 🙏 

If you think it needs testing, give it a `hold`, and I will post the test result images in this PR.